src/main/template/cassandra.yaml

```diff
@@ -0,0 +1,1099 @@
+      clusters, make sure that you give both clusters the same name.
+    Type: String
+    MinLength: 1
+
+  dataCenterNameSuffix:
+    Description: The data center name suffix. By default, the Ec2Snitch names data centers according to the AWS region
+      of the deployment (i.e. "eu-central") and the rack according to the availability zone (i.e. "1a"). Appending a
+      suffix to the data center name allows multiple data centers of the same cluster to run in the same region. If
+      this is not a use case for you, feel free to leave it blank.
+    Type: String
+
+  internalDomain:
+    Description: The internal DNS domain name to be used by the Cassandra cluster. If you run multiple clusters in the
+      same VPC, the internal domain names must be distinct.
+    Type: String
+    MinLength: 1
+    Default: cassandra
+
+  deleteEbsVolumesOnTermination:
+    Description: Whether the attached EBS volumes of the cassandra nodes shall be deleted on termination of the node or
+      not. In production systems, it might be prudent to set this to false, to ensure that data is never destroyed.
+      erroneously.
+    Type: String
+    AllowedValues:
+      - "true"
+      - "false"
+    Default: "true"
+
+Mappings:
+  Scripts:
+    Cassandra:
+      CreateHostsEntry: |
+        #!/bin/bash -xe
+        sed -i "/127.0.0.1/c\127.0.0.1 localhost $(curl http://169.254.169.254/latest/meta-data/local-hostname)" /etc/hosts
+
+      SetupDataDirectories: |
+        #!/bin/bash -xe
+        export CASSANDRA_YAML="/etc/cassandra/conf/cassandra.yaml"
+        mkdir /var/cassandra
+        blkid $(readlink -f /dev/sdf) || mkfs -t xfs $(readlink -f /dev/sdf)
+        mkdir /var/cassandra/data
+        xfs_admin -L cassandra-dt $(readlink -f /dev/sdf)
+        grep -q ^LABEL=cassandra-dt /etc/fstab || echo "LABEL=cassandra-dt /var/cassandra/data xfs defaults,nofail" >> /etc/fstab
+        grep -q "$(readlink -f /dev/sdf) /var/cassandra/data" /proc/mounts || mount /var/cassandra/data
+        blkid $(readlink -f /dev/sdg) || mkfs -t xfs $(readlink -f /dev/sdg)
+        mkdir /var/cassandra/commitlog
+        xfs_admin -L cassandra-cl $(readlink -f /dev/sdg)
+        grep -q ^LABEL=cassandra-cl /etc/fstab || echo "LABEL=cassandra-cl /var/cassandra/commitlog xfs defaults" >> /etc/fstab
+        grep -q "$(readlink -f /dev/sdg) /var/cassandra/commitlog" /proc/mounts || mount /var/cassandra/commitlog
+        chown -R cassandra /var/cassandra
+        sed -i "/^data_file_directories:$/ {N; s|data_file_directories:\n    - /var/lib/cassandra/data|data_file_directories:\n    - /var/cassandra/data|g}" ${CASSANDRA_YAML}
+        sed -i "/commitlog_directory:/c\commitlog_directory: /var/cassandra/commitlog" ${CASSANDRA_YAML}
+
+      UpdateCassandraYaml: |
+        #!/bin/bash -xe
+        export CASSANDRA_YAML="/etc/cassandra/conf/cassandra.yaml"
+        sed -i "/cluster_name:/c\cluster_name: \'${CLUSTER_NAME}\'" ${CASSANDRA_YAML}
```

Comment:
This `sed` stuff works, but it feels a bit too cumbersome and fragile for my taste. How about replacing this with [yq](https://github.com/mikefarah/yq)? It's a single binary without dependencies.

Using it, this line would turn into
```
yq w -i ${CASSANDRA_YAML} cluster_name "${CLUSTER_NAME}"
```

Granted, most properties in `cassandra.yaml` are not complex (e.g. arrays or nested properties), so `yq` can't really shine here. But IMO it would be good to standardize on a tool which can e.g. verify values or add values to arrays.

Reply:
Addendum: to illustrate the fragility, just imagine a developer forgetting/deleting a single colon:

```
sed -i "/trickle_fsync/c\trickle_fsync: true" ${CASSANDRA_YAML}
```
This will turn
```
trickle_fsync: false
trickle_fsync_interval_in_kb: 10240
```

into 

```
trickle_fsync: true
trickle_fsync: true
```